testsuite/classlibrary/TestUGen_RTAlloc.sc

```diff
@@ -70,210 +96,154 @@ TestUGen_RTAlloc : UnitTest {
+		this.assertAllocPass("Convolution2", { Convolution2.ar(DC.ar(1), LocalBuf(256).set(1), 1, 64) });
+		this.assertAllocPass("Convolution2L", { Convolution2L.ar(DC.ar(1), LocalBuf(256).set(1), 1, 64) });
+		this.assertAllocPass("Convolution3", { Convolution3.ar(DC.ar(1), LocalBuf(256).set(1), 1, 64) });
+		this.assertAllocPass("StereoConvolution2L", {
+			var k = LocalBuf(64).set(1); StereoConvolution2L.ar(DC.ar(1), k, k, 1, 64)
+		});
+		this.assertAllocPass("GrainBuf", { GrainBuf.ar(1, 1, 1, LocalBuf(1).set(1), maxGrains: 2)});
+		this.assertAllocPass("GrainFM", { GrainFM.ar(1, 1, 1, maxGrains: 2) });
+		this.assertAllocPass("GrainIn", { GrainIn.ar(1, 1, in:DC.ar(1), maxGrains:2) });
+		this.assertAllocPass("GrainSin", { GrainSin.ar(1, 1, maxGrains: 2) });
 
-		onlyPassUGens.keysValuesDo {|name, func|
-			testAllocPass.value(name.asString, func, 0.001);
-		};
+		this.assertAllocPass("Gendy1", { Gendy1.ar(initCPs: 1024) });
+		this.assertAllocPass("Gendy2", { Gendy2.ar(initCPs: 1024) });
+		this.assertAllocPass("Gendy3", { Gendy3.ar(initCPs: 1024) });
 
-		longerRuntimeUGens.keysValuesDo {|name, func|
-			testAllocPass.value(name.asString, func, 0.02);
-		};
+		this.assertAllocPass("IFFT", { IFFT(LocalBuf(256)) });
 
-		// now fill mem and test allocFail
-		memFill = fillMemFunc.play(server);
+		this.assertAllocPass("SpecPcile", { K2A.ar(SpecPcile.kr(LocalBuf(256))) });
 
-		ugenTests.keysValuesDo {|name, func|
-			testAllocFail.value(name.asString, func, 0.001);
-		};
+		this.assertAllocPass("GVerb", { GVerb.ar(DC.ar(1), 1, maxroomsize: 100) });
 
-		longerRuntimeUGens.keysValuesDo {|name, func|
-			testAllocFail.value(name.asString, func, 0.02);
-		};
+		// -- ONLY PASS UGENS: couldn't reliably test allocFail on these:
+		this.assertAllocPass("IEnvGen", { IEnvGen.ar(Env([1,1]), 0) });
+		this.assertAllocPass("PanAz", { PanAz.ar(12, DC.ar(1)) });
+		this.assertAllocPass("Klank", { Klank.ar(`[100!2, nil, 1], DC.ar(1)) });
+		this.assertAllocPass("Klang", { Klang.ar(`[100!2, nil, 1]) });
+		this.assertAllocPass("Dshuf", { Demand.ar(DC.ar(1),0, Dshuf([1])) });
+		// FFT UGens: only tested for allocPass (hard to make these fail before LocalBuf or FFT fails)
+		this.assertAllocPass("Onsets", { K2A.ar(Onsets.kr(FFT(LocalBuf(128), Impulse.ar(1)))) });
+		this.assertAllocPass("Loudness", { K2A.ar(Loudness.kr(FFT(LocalBuf(128), SinOsc.ar))) });
+		this.assertAllocPass("BeatTrack", { K2A.ar(BeatTrack.kr(FFT(LocalBuf(128)))) });
 
-		memFill.free;
-		// now test PartConv
-		partConvTest.value;
+		// special test for PartConv: requires buffer alloc
+		{
+			var b = Buffer.loadCollection(server, 1!(512));
+			server.sync;
+			this.assertAllocPass("PartConv", { PartConv.ar(DC.ar(1), 256, b) }, 128);
+			b.free;
+		}.value;
+	}
 
+	test_allUGens_allocFail {
+
+		var memSizeKb = server.options.memSize;
+		var memSizeFloats = memSizeKb * 1024 / 4;
+
+		this.assertAllocFail("AllpassC", { AllpassC.ar(DC.ar(1), 100) });
```

Comment:
same as above, make the failing of the test dependent of memSizeFloat, so we could run the tests with different memSizes at some point.